inspirehep/modules/workflows/models.py

```diff
@@ -88,20 +96,31 @@ class WorkflowsRecordSources(db.Model):
         db.PrimaryKeyConstraint('record_id', 'source'),
     )
 
-    source = db.Column(
-        db.Text,
-        default='',
-        nullable=False,
-    )
     record_id = db.Column(
         UUIDType,
         db.ForeignKey('records_metadata.id', ondelete='CASCADE'),
         nullable=False,
     )
+
+    source = db.Column(
+        db.Enum(SourceEnum),
```

Comment:
Autogenerated migrations have problems with enum types. Not sure if this is the case here though...

Reply:
We could use a specific Postgres ENUM data type. Please read it up http://docs.sqlalchemy.org/en/latest/dialects/postgresql.html#sqlalchemy.dialects.postgresql.ENUM and evaluate if it is worth.